code/modules/antagonists/abductor/abductor_structures.dm

```diff
@@ -0,0 +1,123 @@
+		I.play_tool_sound(src)
+		if(I.use_tool(src, user, 30))
+			playsound(src, 'sound/items/deconstruct.ogg', 50, TRUE)
+			for(var/i in 0 to framestackamount)
+				new framestack(get_turf(src))
+			qdel(src)
+			return
+	if(istype(I, /obj/item/stack/sheet/mineral/abductor))
+		var/obj/item/stack/sheet/P = I
+		if(P.get_amount() < 1)
+			to_chat(user, span_warning("You need one alien alloy sheet to do this!"))
+			return
+		to_chat(user, span_notice("You start adding [P] to [src]..."))
+		if(do_after(user, 50, target = src))
+			P.use(1)
+			new /obj/structure/table/abductor(src.loc)
+			qdel(src)
+		return
+	if(istype(I, /obj/item/stack/sheet/mineral/silver))
+		var/obj/item/stack/sheet/P = I
+		if(P.get_amount() < 1)
+			to_chat(user, span_warning("You need one sheet of silver to do this!"))
+			return
+		to_chat(user, span_notice("You start adding [P] to [src]..."))
+		if(do_after(user, 50, target = src))
+			P.use(1)
+			new /obj/structure/table/optable/abductor(src.loc)
+			qdel(src)
+
+/obj/structure/table/abductor
+	name = "alien table"
+	desc = "Advanced flat surface technology at work!"
+	icon = 'icons/obj/smooth_structures/alien_table.dmi'
+	icon_state = "alien_table-0"
+	base_icon_state = "alien_table"
+	buildstack = /obj/item/stack/sheet/mineral/abductor
+	framestack = /obj/item/stack/sheet/mineral/abductor
+	buildstackamount = 1
+	framestackamount = 1
+	smoothing_groups = SMOOTH_GROUP_ABDUCTOR_TABLES
+	canSmoothWith = SMOOTH_GROUP_ABDUCTOR_TABLES
+	frame = /obj/structure/table_frame/abductor
+	custom_materials = list(/datum/material/silver =SHEET_MATERIAL_AMOUNT)
+
+/obj/structure/table/optable/abductor
+	name = "alien operating table"
+	desc = "Used for alien medical procedures. The surface is covered in tiny spines."
+	frame = /obj/structure/table_frame/abductor
+	buildstack = /obj/item/stack/sheet/mineral/silver
+	framestack = /obj/item/stack/sheet/mineral/abductor
+	buildstackamount = 1
+	framestackamount = 1
+	icon = 'icons/obj/abductor.dmi'
+	icon_state = "bed"
+	can_buckle = TRUE
+	buckle_lying = 90
+	/// Amount to inject per second
+	var/inject_am = 0.5
+
+	var/static/list/injected_reagents = list(/datum/reagent/medicine/cordiolis_hepatico)
+
+/obj/structure/table/optable/abductor/Initialize(mapload)
+	. = ..()
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERED = PROC_REF(on_entered),
+	)
+	AddElement(/datum/element/connect_loc, loc_connections)
+
+/obj/structure/table/optable/abductor/proc/on_entered(datum/source, atom/movable/AM)
+	SIGNAL_HANDLER
+	if(iscarbon(AM))
+		START_PROCESSING(SSobj, src)
+		to_chat(AM, span_danger("You feel a series of tiny pricks!"))
+
+/obj/structure/table/optable/abductor/process(seconds_per_tick)
+	. = PROCESS_KILL
+	for(var/mob/living/carbon/C in get_turf(src))
+		. = TRUE
+		for(var/chemical in injected_reagents)
+			if(C.reagents.get_reagent_amount(chemical) < inject_am * seconds_per_tick)
+				C.reagents.add_reagent(chemical, inject_am * seconds_per_tick)
```

Comment:
```suggestion
				C.reagents.add_reagent(chemical, inject_am * seconds_per_tick)
				
	return .
```

it's probably okay to have the funny dot operator value for runtime safety in reagent code but you must pay the dot operator toll (explicity goodness)